src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -345,17 +351,18 @@ private Future<String> handleLogin(LoginCredentials credentials, String tenantId
     return promise.future();
   }
 
-  private void failWithInternalError(Promise<String> promise, Handler<AsyncResult<Response>> asyncResultHandler,
+  private void failWithInternalError(Promise<TenantContext> promise, Handler<AsyncResult<Response>> asyncResultHandler,
                                      Throwable cause) {
     logger.error(cause);
     asyncResultHandler.handle(Future.succeededFuture(PostAuthnLoginResponse.respond500WithTextPlain(
         INTERNAL_ERROR)));
     promise.fail(cause);
   }
 
-  private void loginAfterUserVerified(AsyncResult<JsonObject> verifyResult, LoginCredentials entity, String tenantId,
+  private void loginAfterUserVerified(AsyncResult<JsonObject> verifyResult, LoginCredentials entity, TenantContext newTenantContext,
```

Comment:
Lets rename just to tenantContext

Reply:
due to decided to change the approach, this comment isn't actual